pkg/epp/scheduling/plugins/scorer/queue.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scorer
+
+import (
+	"math"
+
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/scheduling/types"
+)
+
+type QueueScorer struct {
+}
+
+func (q *QueueScorer) Name() string {
+	return "queue"
+}
+
+func (q *QueueScorer) Score(ctx *types.SchedulingContext, pods []types.Pod) map[types.Pod]float64 {
+	minQueueSize := math.MaxInt
+	maxQueueSize := math.MinInt
+
+	// Iterate through the remaining pods to find min and max
+	for _, pod := range pods {
+		queueSize := pod.GetMetrics().WaitingQueueSize
+		if queueSize < minQueueSize {
+			minQueueSize = queueSize
+		}
+		if queueSize > maxQueueSize {
+			maxQueueSize = queueSize
+		}
+	}
+
+	// podScoreFunc calculates the score based on the queue size of each pod. Longer queue gets a lower score.
+	podScoreFunc := func(ctx *types.SchedulingContext, pod types.Pod) float64 {
+		if maxQueueSize == 0 || maxQueueSize == minQueueSize {
+			// If all pods have the same queue size, return a neutral score
+			return 1.0
+		}
+		return float64(maxQueueSize-pod.GetMetrics().WaitingQueueSize) / float64(maxQueueSize-minQueueSize)
+	}
+
+	return ToScoreFunc(podScoreFunc)(ctx, pods)
+}
```

Comment:
if we had the NormalizeScore function in the scheduler similarly to [kube-scheduler logic](https://kubernetes.io/docs/concepts/scheduling-eviction/scheduling-framework/#normalize-scoring), would this scorer look simpler?
the max-min calculations looks like score normalization.

Reply:
The complexity in this scorer really comes from the calculation of the max and min queue size. If we use a normalizeScore, then we simply shift the complexity to the normalizeScore I think?

IMO having another normalize score plugin adds more cognitive overhead. To me having Score and weights is very straightforward: 0-least preferred, 1-most preferred.

Reply:
k, was just wondering about your opinion here :)

Reply:
We may need something like this in the future if we find a pattern across many scorer plugins. Honestly for what we have now, I don't think it's worth it.  It took me quite some time to read the scheduler NormalizeScore implementations.

Reply:
have we established that scorers must return a value in a specific range? In kube-scheduler we set that to 0-100 IIRC, not 0-1.

Reply:
Yes it's [0, 1] https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/a6ee5590dc54fc3018a4e5951592dcf2a7b25c46/pkg/epp/scheduling/plugins/plugins.go#L53